hipposerve/api/product.py

```diff
@@ -274,6 +277,60 @@ async def confirm_product(
     logger.info("Successfully confirmed product {} (id: {})", item.name, item.id)
 
 
+@product_router.get("/{id}/set-visibility/{visibility}")
+async def set_visibility(
+    id: PydanticObjectId,
+    visibility: str,
+    calling_user: UserDependency,
```

Comment:
This should not be a GET request and visibility should not be a parameter in the URL. This is a classic example of a scenario that should use the POST method and include visibility as a body parameter.